### PR TITLE
Feature hospital database

### PR DESCRIPTION
**Pull request by HenaryKhuraijam**
## Database based on a diagram
- Added database diagram for a clinic.
- Implemented the database from the diagram.

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -0,0 +1,27 @@
+CREATE TABLE medical_histories (id INT PRIMARY KEY, admitted_at TIMESTAMP, status VARCHAR(200), patient_id INT);
+
+CREATE TABLE patients (id INT PRIMARY KEY, name VARCHAR(100), date_of_birth DATE);
+
+CREATE TABLE invoices (id int PRIMARY KEY, total_amount DECIMAL, generated_at TIMESTAMP, payed_at TIMESTAMP, medical_histories_id INT);
+
+CREATE TABLE invoice_items (id INT PRIMARY KEY, unit_price DECIMAL, quantity INT, total_price DECIMAL, invoices_id INT, treatments_id INT);
+
+CREATE TABLE treatments (id INT PRIMARY KEY, type VARCHAR(100), name VARCHAR(100));
```
Comment (Munsa1) [inline schema.sql]: - [ ] Good job implementing this project, however the new tables should be created in a file called `schema_based_on_diagram.sql` file. Kindly rename this file to this name as per the requirement.

File: schema.sql [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+ALTER TABLE invoice_items ADD FOREIGN KEY (invoices_id) REFERENCES invoices(id);
+
+ALTER TABLE invoice_items ADD FOREIGN KEY (treatments_id) REFERENCES treatments(id);
+
+CREATE TABLE treatments_histories (medical_histories_id INT, treatments_id INT);
+
+ALTER TABLE treatments_histories ADD FOREIGN KEY (medical_histories_id) REFERENCES medical_histories(id);
+
+ALTER TABLE treatments_histories ADD FOREIGN KEY (treatments_id) REFERENCES treatments(id);
```
Comment (Munsa1) [inline schema.sql]: - [ ] As per requirement please add `INDEXES` for all your foreign keys to meet the requirements.

File: README.md [hunk 3]
```diff
@@ -1 +1,148 @@
...
+<!-- CONTRIBUTING -->
+
+## 🤝 Contributing <a name="contributing"></a>
+
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](https://github.com/jubaer919/hospital-database.git/issues).
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
```
Comment (Munsa1) [inline README.md]: - [ ] Kindly fix the broken link to the issues page for a more professional readme file.

File: README.md [hunk 4]
```diff
@@ -1 +1,148 @@
...
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENSE.md) licensed._
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
```
Comment (Munsa1) [inline README.md]: - [ ] Fix the broken MIT license link for a more professional readme file. Add the License to your project and link it to your readme file.

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @HenaryKhuraijam @jubaer919 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Created All Tables In The Given Schema :heavy_check_mark: 
Created All Foreign Keys For All Tables :heavy_check_mark: 
 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 2 above)
Comment (jubaer919) [reply schema.sql]: do i need to create index for foreign key

File: schema_based_on_diagram.sql [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+CREATE INDEX medical_histories_id_asc ON invoices (medical_histories_id ASC);
+
+CREATE INDEX invoices_id_asc ON  invoice_items (invoices_id ASC);
+
+CREATE INDEX treatments_id_asc ON invoice_items (treatments_id ASC);
+
+CREATE INDEX treatments_id_asc ON treatments_histories (treatments_id ASC);
+
+CREATE INDEX medical_histories_id_asc ON treatments_histories (medical_histories_id ASC);
```
Comment (shahadat3669) [inline schema_based_on_diagram.sql:39]: - [ ] Good job on adding indexes for all foreign keys as mentioned by the previous reviewer's request. However, I noticed that the names for two of the indexes conflict with existing indexes, resulting in an error like `ERROR: relation "medical_histories_id_asc" already exists (SQLSTATE 42P07)`. To resolve this, kindly review your code and provide unique names for these indexes. Run the index creation again to ensure they work without errors. 🏷️🔄🐘
image

**Review by shahadat3669 — APPROVED** [review]
Hi @HenaryKhuraijam, @jubaer919, 

Your project is complete!  There is nothing else to say other than it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ new tables are created in schema_based_on_diagram.sql file.
✅ all "join table" for the many-to-many relastionships is created
✅ Both partner worked

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 2 above)
Comment (Munsa1) [reply schema.sql]: Yes

**Pull request merged**